src/js/components/Breadcrumbs.js

```diff
@@ -0,0 +1,242 @@
+
+    return textWidth + PADDED_ICON_WIDTH;
+  }
+
+  getExpandedWidth() {
+    // array/splat casts NodeList to array
+    let listItems = [...ReactDOM.findDOMNode(this).children]
+      .filter(function (_, index) {
+        // Filter out even nodes containing '>'
+        return index % 2;
+      });
+
+    return listItems
+      .map((item, index) => {
+        let isFirstItem = index === 0;
+        let isLastItem = index === listItems.length - 1;
+        if (isFirstItem || isLastItem) {
+          return this.getWidthFromExpandedItem(item);
+        }
+
+        return this.getWidthFromCollapsedItem(item);
+      })
+      .reduce(
+        function (totalWidth, itemWidth) { return totalWidth + itemWidth },
+        this.getLastItemWidth()
+      );
+  }
+
+  getLastItemWidth() {
+    let lastItem = ReactDOM.findDOMNode(this).lastChild;
+    let lastItemLink = lastItem.firstChild;
+
+    return lastItemLink.scrollWidth + LAST_ITEM_OFFSET;
+  }
+
+  getCurrentRouteParams() {
+    // Isolated for easier testing
+    return this.context.router.getCurrentParams();
+  }
+
+  wrapListItem(crumb, key) {
+    return (
+      <li key={key}>
+        {crumb}
+      </li>
+    );
+  }
+
+  getCrumb(crumb, key) {
+    if (!React.isValidElement(crumb)) {
+      if (typeof crumb === 'string') {
+        crumb = {label: crumb};
+      }
+
+      crumb = (<Crumb {...crumb} />);
+    }
+
+    return this.wrapListItem(crumb, key);
+  }
+
+  getCrumbDivider(key) {
+    return (
+      <li key={key} >
+        <IconChevron
+          className="icon icon-small"
+          isForward={true} />
+      </li>
+    );
+  }
+
+  shouldCollapse(availableWidth, expandedWidth) {
+    // Smooth collapse action to prevent flickering between states
+    if (this.state.collapsed) {
+      return expandedWidth >= availableWidth - COLLAPSE_BUFFER;
+    }
+
+    return expandedWidth >= availableWidth + COLLAPSE_BUFFER;
+  }
+
+  buildCrumbs(routeName) {
+    let {router} = this.context;
+    let {namedRoutes} = router;
+    let route = namedRoutes[routeName];
+
+    if (!route || !route.buildBreadCrumb) {
+      return [];
+    }
+
+    let crumbConfiguration = route.buildBreadCrumb();
+    let crumbs = crumbConfiguration.getCrumbs(router);
+
+    if (crumbConfiguration.parentCrumb) {
+      crumbs = this.buildCrumbs(crumbConfiguration.parentCrumb).concat(crumbs);
+    }
+
+    return crumbs;
+  }
+
+  renderCrumbsFromRoute() {
+    let {router} = this.context;
+    let routes = router.getCurrentRoutes();
+    let currentRoute = routes[routes.length - 1];
+    let crumbs = this.buildCrumbs(currentRoute.name);
+
+    crumbs = crumbs.slice(this.props.shift);
+
+    let crumbKey = 0;
+
+    return crumbs.reduce((memo, crumb, crumbIndex) => {
+      memo.push(this.getCrumb(crumb, ++crumbKey));
+
+      if (crumbs.length - 1 !== crumbIndex) {
+        memo.push(this.getCrumbDivider(++crumbKey));
+      }
+
+      return memo;
+    }, []);
+  }
```

Comment:
This method and `buildCrumbs` are the ones that have changed from previous implementation, everything else should be the same.